util/rlp/src/lib.rs

```diff
@@ -108,6 +108,12 @@ pub fn encode<E>(object: &E) -> ElasticArray1024<u8> where E: Encodable {
 	stream.drain()
 }
 
+pub fn encode_short<E>(object: &E) -> ElasticArray8<u8> where E: Encodable {
```

Comment:
would really like it to be something like `rlp::encode_into<E: Encodable, U: Growable + DerefMut<[u8]>>` where that trait is implemented for vecs, all elastic arrays, etc.

Extend `Encodable` trait with a `type Encoded: Growable + DerefMut<[u8]>>` and `rlp::encode<E: Encodable>(e: &E) -> E::Encoded` and we can easily encode a type into the exact amount of space it needs without thinking about it.

Reply:
great idea! but it requires much more modifications than changing signature of a method. I'll create an issue to not forget about it

Reply:
Why not start with `ElasticArray8<u8>` or even `ElasticArray1<u8>` and then switch to `ElasticArray1024` if it's not enough? The return type could probably be an enum with required conversions / Derefs.

Reply:
if it's an enum it will take even more stack space than the largest variant. although since it can leave the end uninitialized for smaller variants it might still be faster than using 1024 for everything.

Reply:
exactly!

Reply:
I don't think so. I'm pretty sure the `ElasticArray1024` has to zero all of its internal buffer before use.